frame/vault/src/tests.rs

```diff
@@ -525,93 +525,125 @@ proptest! {
+				&vault_id,
+				&strategy_account_id,
+				&strategy_vault_balance
+			));
 
-			// Shareholders total LP
-			let lp_total = Tokens::total_issuance(vault.lp_token_id);
+			let lp_tokens_total = Tokens::total_issuance(vault.lp_token_id);
 
-			// Users depositing later
-			for (account, balance) in accounts[strategy_deposit_moment..accounts.len()].iter().copied() {
-				prop_assert_ok!(
-					Vaults::deposit(Origin::signed(account), vault_id, balance)
-				);
+			// Liquidity providers deposit all their native tokens to receive LP tokens
+			// AFTER losses and profits
+			for (account, initial_native_tokens) in after_moment_lps() {
+				let origin = Origin::signed(account);
+				prop_assert_ok!(Vaults::deposit(origin, vault_id, initial_native_tokens));
 			}
 
-			// Withdraw & local check
-			for ((account, balance), index) in accounts.iter().copied().zip(account_start..) {
-				// Current lp
-				let lp = Tokens::balance(vault.lp_token_id, &account);
-
-				// Withdraw all my shares, including profits
-				prop_assert_ok!(Vaults::withdraw(Origin::signed(account), vault_id, lp));
-
-				// Balance after having deposited + withdrawn my funds
-				let new_balance = Tokens::balance(asset_id, &account);
-
-				// We had shares before the profit, we get a cut of the profit
-				if index <= strategy_deposit_moment {
-					// Compute my share
-					let strategy_profit_share =
-						multiply_by_rational(strategy_profits, lp, lp_total).expect("qed;");
-
-					prop_assert_epsilon!(new_balance, balance + strategy_profit_share);
-				}
-				else {
+			for (idx, (account, initial_native_tokens)) in lps.iter().copied().enumerate() {
+				//  Contains half of LP balances minus LP profits
+				let half_initial_native_tokens = initial_native_tokens / 2;
+
+				let lp_tokens = Tokens::balance(vault.lp_token_id, &account);
+				// Because of `<Vaults as StrategicVault>::withdraw`, the vault does not own 100% of
+				// the funds. Therefore, a full withdraw is not possible.
+				let withdrawn_lp_tokens = lp_tokens / 2;
+
+				// Withdraws all LP tokens
+				prop_assert_ok!(Vaults::withdraw(Origin::signed(account), vault_id, withdrawn_lp_tokens));
+
+				// New balance that includes losses and profits
+				let new_native_tokens = Tokens::balance(asset_id, &account);
+
+				let curr_lp_deposited_before_moment = lps_start + idx <= strategy_moment;
+
+				if curr_lp_deposited_before_moment {
+					let strategy_native_tokens_deposit = multiply_by_rational(
+						strategy_deposit / 2,
+						lp_tokens,
+						lp_tokens_total,
+					)
+					.expect("qed;");
+					let strategy_native_tokens_withdraw = multiply_by_rational(
+						strategy_withdraw_diff / 2,
+						lp_tokens,
+						lp_tokens_total,
+					)
+					.expect("qed;");
+
+					let diff = strategy_native_tokens_deposit - strategy_native_tokens_withdraw;
+
+					prop_assert_epsilon!(new_native_tokens, half_initial_native_tokens + diff);
```

Comment:
Works as expected because the simply removal of, for example, withdrawn tokens would result in an checking error